File: nodejs-express-api/models/marks.js

import { BaseModel, sequelize, Sequelize } from "./basemodel.js";

class Marks extends BaseModel {
	static init() {
		return super.init(
			{
				
				id: { type: Sequelize.INTEGER, primaryKey: true, autoIncrement: true },
				year: { type:Sequelize.STRING , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				term: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				uid: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				maths: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				english: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				kiswahili: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				chemistry: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				biology: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				physics: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				history: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				geography: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				cre: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				business: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				computer: { type:Sequelize.INTEGER , allowNull: false ,defaultValue: Sequelize.literal('DEFAULT') },
				form: { type:Sequelize.STRING   }
			}, 
			{ 
				sequelize,
				
				tableName: "marks",
				modelName: "marks",
			}
		);
	}
	
	static listFields() {
		return [
			Sequelize.literal('marks.year AS year'), 
			Sequelize.literal('marks.term AS term'), 
			Sequelize.literal('marks.form AS form'), 
			Sequelize.literal('marks.uid AS uid'), 
			Sequelize.literal('marks.maths AS maths'), 
			Sequelize.literal('marks.english AS english'), 
			Sequelize.literal('marks.kiswahili AS kiswahili'), 
			Sequelize.literal('marks.chemistry AS chemistry'), 
			Sequelize.literal('marks.biology AS biology'), 
			Sequelize.literal('marks.physics AS physics'), 
			Sequelize.literal('marks.history AS history'), 
			Sequelize.literal('marks.geography AS geography'), 
			Sequelize.literal('marks.cre AS cre'), 
			Sequelize.literal('marks.business AS business'), 
			Sequelize.literal('marks.computer AS computer'), 
			Sequelize.literal('marks.id AS id')
		];
	}

	static viewFields() {
		return [
			'id', 
			'year', 
			'term', 
			'uid', 
			'maths', 
			'english', 
			'kiswahili', 
			'chemistry', 
			'biology', 
			'physics', 
			'history', 
			'geography', 
			'cre', 
			'business', 
			'computer', 
			'form'
		];
	}

	static editFields() {
		return [
			'year', 
			'term', 
			'uid', 
			'form', 
			'maths', 
			'english', 
			'kiswahili', 
			'chemistry', 
			'biology', 
			'physics', 
			'history', 
			'geography', 
			'cre', 
			'business', 
			'computer', 
			'id'
		];
	}

	
	static searchFields(){
		return [
			Sequelize.literal("marks.year LIKE :search"), 
			Sequelize.literal("marks.form LIKE :search"), 
			Sequelize.literal("marks.uid LIKE :search"), 
			Sequelize.literal("marks.id LIKE :search"),
		];
	}

	
	
}
Marks.init();
export default Marks;

File: python-flask-api/app/models/marks.py
from app import db
from datetime import datetime
from wtforms import Form, StringField
from wtforms.validators import *
from .basemodel import BaseModel
from ..helpers.utils import OptionalButNotEmpty
from ..helpers.utils import OptionalNullable


class Marks(db.Model):
    __tablename__ = 'marks'
    id = db.Column("id", db.Integer, primary_key=True, autoincrement=True)
    year = db.Column("year", db.String)
    term = db.Column("term", db.Integer)
    uid = db.Column("uid", db.Integer)
    maths = db.Column("maths", db.Integer)
    english = db.Column("english", db.Integer)
    kiswahili = db.Column("kiswahili", db.Integer)
    chemistry = db.Column("chemistry", db.Integer)
    biology = db.Column("biology", db.Integer)
    physics = db.Column("physics", db.Integer)
    history = db.Column("history", db.Integer)
    geography = db.Column("geography", db.Integer)
    cre = db.Column("cre", db.Integer)
    business = db.Column("business", db.Integer)
    computer = db.Column("computer", db.Integer)
    form = db.Column("form", db.String)
    
    def _asdict(self):
       return {c.name: getattr(self, c.name) for c in self.__table__.columns}


class MarksAddForm(Form):
    year = StringField('Year', [InputRequired()])
    term = StringField('Term', [InputRequired()])
    uid = StringField('Uid', [InputRequired()])
    form = StringField('Form', [OptionalNullable()])
    maths = StringField('Maths', [InputRequired()])
    english = StringField('English', [InputRequired()])
    kiswahili = StringField('Kiswahili', [InputRequired()])
    chemistry = StringField('Chemistry', [InputRequired()])
    biology = StringField('Biology', [InputRequired()])
    physics = StringField('Physics', [InputRequired()])
    history = StringField('History', [InputRequired()])
    geography = StringField('Geography', [InputRequired()])
    cre = StringField('Cre', [InputRequired()])
    business = StringField('Business', [InputRequired()])
    computer = StringField('Computer', [InputRequired()])


class MarksEditForm(Form):
    year = StringField('Year', [OptionalButNotEmpty()])
    term = StringField('Term', [OptionalButNotEmpty()])
    uid = StringField('Uid', [OptionalButNotEmpty()])
    form = StringField('Form', [OptionalNullable()])
    maths = StringField('Maths', [OptionalButNotEmpty()])
    english = StringField('English', [OptionalButNotEmpty()])
    kiswahili = StringField('Kiswahili', [OptionalButNotEmpty()])
    chemistry = StringField('Chemistry', [OptionalButNotEmpty()])
    biology = StringField('Biology', [OptionalButNotEmpty()])
    physics = StringField('Physics', [OptionalButNotEmpty()])
    history = StringField('History', [OptionalButNotEmpty()])
    geography = StringField('Geography', [OptionalButNotEmpty()])
    cre = StringField('Cre', [OptionalButNotEmpty()])
    business = StringField('Business', [OptionalButNotEmpty()])
    computer = StringField('Computer', [OptionalButNotEmpty()])
